common/lib/capa/capa/safe_exec/helpers.py

```diff
@@ -0,0 +1,92 @@
+"""
+Helper methods related to safe exec.
+"""
+
+import requests
+import json
+import logging
+
+from codejail.safe_exec import SafeExecException
+from django.conf import settings
+from django.utils.translation import ugettext as _
+from edx_toggles.toggles import SettingDictToggle
+from requests.exceptions import RequestException, HTTPError
+from simplejson import JSONDecodeError
+
+from .exceptions import CodejailServiceParseError, CodejailServiceStatusError, CodejailServiceUnavailable
+
+log = logging.getLogger(__name__)
+
+# .. toggle_name: ENABLE_CODEJAIL_REST_SERVICE
+# .. toggle_implementation: DjangoSetting
+# .. toggle_default: False
+# .. toggle_description: Set this to True if you want to run Codejail code using
+#   a separate VM or container and communicate with edx-platform using REST API.
+# .. toggle_use_cases: open_edx
+# .. toggle_creation_date: 2021-08-19
+# .. toggle_target_removal_date: None
+# .. toggle_warnings:
+# .. toggle_tickets:
+ENABLE_CODEJAIL_REST_SERVICE = SettingDictToggle(
+    "FEATURES", "ENABLE_CODEJAIL_REST_SERVICE", default=False, module_name=__name__
+)
```

Comment:
I think we want to add new settings to the top level rather than `FEATURES`. (There isn't an ADR on this, as far as I know, but it seems like FEATURES is just an older approach.)
```suggestion
ENABLE_CODEJAIL_REST_SERVICE = SettingToggle(
    "ENABLE_CODEJAIL_REST_SERVICE", default=False, module_name=__name__
)
```

Reply:
@timmc-edx In that case I would need to remove this right https://github.com/edx/edx-platform/blob/4aef32eb717ad1cd935a32e844ff9b0e3badd472/lms/envs/common.py#L952 ? Do I still need to add ENABLE_CODEJAIL_REST_SERVICE in the env files ?

Reply:
That's correct, it should be removed from the envs file as well.